File: frontend/demo/component/login/login-overlay-custom-form-area.ts
import 'Frontend/demo/init'; // hidden-source-line
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import '@vaadin/integer-field';
import '@vaadin/login';
import { applyTheme } from 'Frontend/generated/theme';

@customElement('login-overlay-custom-form-area')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <!-- no-autofocus is used to prevent the example from stealing focus when browsing the documentation -->
      <vaadin-login-overlay opened no-autofocus>
        <vaadin-integer-field
          slot="custom-form-area"
          name="code"
          label="One-time code"
        ></vaadin-integer-field>
      </vaadin-login-overlay>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/login/LoginOverlayCustomFormArea.java
package com.vaadin.demo.component.login;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.login.LoginOverlay;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.textfield.IntegerField;
import com.vaadin.flow.router.Route;

@Route("login-overlay-custom-form-area")
public class LoginOverlayCustomFormArea extends Div {

    public LoginOverlayCustomFormArea() {
        // tag::snippet[]
        LoginOverlay loginOverlay = new LoginOverlay();
        IntegerField code = new IntegerField("One-time code");
        code.getElement().setAttribute("name", "code");
        loginOverlay.getCustomFormArea().add(code);
        // end::snippet[]
        add(loginOverlay);
        loginOverlay.setOpened(true);
        // Prevent the example from stealing focus when browsing the
        // documentation
        loginOverlay.getElement().setAttribute("no-autofocus", "");
    }

    public static class Exporter extends DemoExporter<LoginOverlayCustomFormArea> { // hidden-source-line
    } // hidden-source-line
}
